packages/@lwc/synthetic-shadow/src/polyfills/event-target/polyfill.ts

```diff
@@ -19,22 +19,26 @@ import { getEventListenerWrapper } from '../../shared/event-target';
 
 function patchedAddEventListener(
     this: EventTarget,
-    _type: string,
-    _listener: EventListenerOrEventListenerObject,
-    _optionsOrCapture?: boolean | AddEventListenerOptions
+    type: string,
+    listener: EventListenerOrEventListenerObject,
+    optionsOrCapture?: boolean | AddEventListenerOptions
 ) {
     if (isHostElement(this)) {
         // Typescript does not like it when you treat the `arguments` object as an array
         // @ts-ignore type-mismatch
         return addCustomElementEventListener.apply(this, arguments);
     }
-    const args = ArraySlice.call(arguments);
```

Comment:
We have been using this `ArraySlice.call(arguments)` pattern throughout the entire synthetic shadow code base (20 occurrences). It would be strange to only modify this in a single place.

Keep me in check @ekashida but we are trading off performance here for validity:
- arguments checking is delegated to the browser.
- future-proofness by not accepting a predefined number of arguments. In the case where one of the patched APIs accepts a new argument (spec change), we don't have to update the code for the API to keep working.

Reply:
True, I didn't make this change everywhere. E.g. `removeEventListener` is not called as much as `addEventListener`, so it didn't seem worth it.

I am happy to back out this change if we feel that the 1%-2% improvement is not worth the potential maintenance cost. But I think 1%-2% across 5 benchmarks would make me lean towards taking the improvement.

Reply:
Yeah, we do this because we don't want to handle validation. iirc, different browsers have different validation rules and we didn't want to get into the business of normalizing browser APIs unnecessarily. I am not against making this change, but I think it requires some investigation and maybe integration with WPT?

Reply:
I could also modify this PR to just delegate to the browser in the `arguments.length < 2` case. This is really an edge case that is never called in any of the benchmarks (and shouldn't be called in real-world code), so I don't mind if it uses the slow path.